File: backend/data/responses/fair_value_response.py
"""
Pydantic models for fair value API responses.
"""

from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class FairValueResponse(BaseModel):
    id: int
    symbol: str
    released_at: Optional[datetime]
    source: Optional[str]
    source_ar: Optional[str]
    recommendation: Optional[str]
    recommendation_ar: Optional[str]
    market: Optional[str]
    sector: Optional[str]
    market_url: Optional[str]
    value: Optional[float]
    price: Optional[float]
    last_price: Optional[float]
    change: Optional[float]
    change_percentage: Optional[float]

File: frontend.Old/src/responses/FairValueResponse.ts
export interface FairValueResponse {
  id: number;
  symbol: string;
  released_at?: string;
  source?: string;
  source_ar?: string;
  recommendation?: string;
  recommendation_ar?: string;
  market?: string;
  sector?: string;
  market_url?: string;
  value?: number;
  price?: number;
  last_price?: number;
  change?: number;
  change_percentage?: number;
}

